linetools/spectra/xspectrum1d.py

```diff
@@ -243,9 +252,10 @@ def wavelength(self, value):
     def flux(self):
         """ Return the flux with units
         """
-        flux =  self.data[self.select]['flux'] * self.units['flux']
+        #flux =  self.data[self.select]['flux'] * self.units['flux']
+        flux =  self.data['flux'][self.select].compressed() * self.units['flux']
```

Comment:
Maybe in this case instead of the compressed data, retrieve the masked values as zeros instead?